### PR TITLE
docs(operators): mergeScan and toPromise

### PR DESCRIPTION
**Pull request by gusvargas**
**Description:**
Add docs for `mergeScan` and `toPromise`

### CODE REVIEW

File: src/operator/toPromise.ts [hunk 1]
```diff
@@ -7,8 +7,51 @@ export function toPromise<T>(this: Observable<T>, PromiseCtor: typeof Promise):
...
+ * source
+ *   .then((value) => console.log('Value: %s', value))
+ *   .catch((err) => console.log('Error: %s', err));
+ * // => Error: Error: woops
+ *
+ * // Setting via the config
+ * Rx.config.Promise = RSVP.Promise;
+ *
+ * let source = Rx.Observable
+ *   .just(42)
```
Comment (mattpodwysocki) [inline src/operator/toPromise.ts]: `Rx.Observable.just` does not exist, use `Rx.Observable.of` instead

Comment (gusvargas) [inline src/operator/toPromise.ts]: Ah right. The [examples](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/topromise.md) from the RxJS 4 docs were pretty thorough so I copy/pasted them and forgot to fix this. Fixed in https://github.com/ReactiveX/rxjs/pull/2208/commits/aede7387134f4d1b9cd83f603726775f06475e11.

File: src/operator/mergeScan.ts [hunk 2]
```diff
@@ -9,28 +9,45 @@ import { OuterSubscriber } from '../OuterSubscriber';
...
- * @param concurrent
- * @return {Observable<R>|WebSocketSubject<T>|Observable<T>}
+ * Applies an accumulator function over the source Observable where the
+ * accumulator function itself returns an Observable, then each intermediate
+ * Observable returned is merged into the output Observable.
+ *
+ * <span class="informal">It's like {@link scan}, but the Observables returned
+ * by the accumulator are merged into the outer Observable.</span>
+ *
+ * @example <caption>Count the number of click events</caption>
```
Comment (gusvargas) [inline src/operator/mergeScan.ts]: Open for suggestions for a less contrived example. Here's a thought:

```javascript
const clicks = Observable.fromEvent(
  document.querySelector('button'),
  'click'
);

clicks
  .map((_, i) => i)
  .mergeScan((acc, attempts) => {
    return Observable
      .timer(BACKOFF_MULTIPLIER * attempts)
      .mapTo(attempts + 1);
  }, 0)
  .subscribe(v => {
    console.log(v);
  });
```
http://jsbin.com/dumanuyuxa/1/edit?js,console,output

Still a bit contrived without good context...

File: src/operator/mergeScan.ts [hunk 3]
```diff
@@ -9,28 +9,45 @@ import { OuterSubscriber } from '../OuterSubscriber';
...
+ * var clicks = Rx.Observable.fromEvent(document, 'click');
```
Comment (blesh) [inline src/operator/mergeScan.ts]: have a look at some of the other examples, but I feel like we've been doing examples in TypeScript, and using `let` or `const` in lieu of `var`. Also, it's likely we're naming observables with Finnish Notation like `click$`.

File: src/operator/mergeScan.ts [hunk 4]
```diff
@@ -9,28 +9,45 @@ import { OuterSubscriber } from '../OuterSubscriber';
...
  * @method mergeScan
  * @owner Observable
  */
-export function mergeScan<T, R>(this: Observable<T>, project: (acc: R, value: T) => Observable<R>,
+export function mergeScan<T, R>(this: Observable<T>,
+                                accumulator: (acc: R, value: T) => Observable<R>,
                                 seed: R,
                                 concurrent: number = Number.POSITIVE_INFINITY): Observable<R> {
-  return this.lift(new MergeScanOperator(project, seed, concurrent));
+  return this.lift(new MergeScanOperator(accumulator, seed, concurrent));
```
Comment (blesh) [inline src/operator/mergeScan.ts]: Solid catch, thank you.

File: src/operator/mergeScan.ts [hunk 5]
```diff
@@ -9,28 +9,45 @@ import { OuterSubscriber } from '../OuterSubscriber';
...
+ * <span class="informal">It's like {@link scan}, but the Observables returned
+ * by the accumulator are merged into the outer Observable.</span>
+ *
+ * @example <caption>Count the number of click events</caption>
+ * var clicks = Rx.Observable.fromEvent(document, 'click');
+ * var ones = clicks.mapTo(1);
+ * var seed = 0;
+ * var count = ones.mergeScan((acc, one) => Observable.of(acc + one), seed);
+ * count.subscribe(x => console.log(x));
+ *
```
Comment (blesh) [inline src/operator/mergeScan.ts]: Can you add a comment that illustrates what the expected output is?

Comment (blesh) [inline src/operator/mergeScan.ts]: Again, I think you can look at other docs for examples of this.

File: src/operator/mergeScan.ts (hunk 3 above)
Comment (gusvargas) [inline src/operator/mergeScan.ts]: FYI, I couldn't find any operator examples that are using TypeScript. The only TypeScript I found in the docs were interfaces for some classes, e.g., [Action](http://reactivex.io/rxjs/class/es6/scheduler/Action.js~Action.html). Is this just a change going forward? I'm happy to make that change but just want to confirm.

File: src/operator/mergeScan.ts (hunk 5 above)
Comment (gusvargas) [inline src/operator/mergeScan.ts]: Sure. Did you see my comment above? i.e., any better example ideas for `mergeScan`?

File: src/operator/mergeScan.ts (hunk 3 above)
Comment (gusvargas) [inline src/operator/mergeScan.ts]: To be clear, I didn't see usage of any TypeScript specific language features in the operator examples.

**Comment by gusvargas** [comment]
@blesh this should be ready to go. anything else?

**Comment by blesh** [comment]
@gusvargas ... nope... it's just slow over the holidays. These things all need to be reviewed still. Thank you for your contribution.

**Comment by staltz** [comment]
Looks rather Ok to me, although the toPromise JSDoc could be made to look like other docs. It's quite hard to document mergeScan and good job finding an example for it.

**Comment by gusvargas** [comment]
@staltz What exactly would you like more aligned with the other docs?

It seems like most docs are more or less organized as

1. Overview description
2. Informal tip if applicable
3. Marble diagram
4. More in-depth explanation
5. Example
6. Interface

I'm missing 2, 3, and 4. I couldn't think of anything more insightful to add for 2 and 4. Did you want me to add a marble diagram?

**Pull request merged**